### DISCUSSION TITLE
Want Push notification make customizable every statge

### DISCUSSION THREAD

**Discussion by LoganRaja** [invertase/react-native-firebase#4785] [Q&A] [answered]
I am working on push notification, I replace all stages (background, quit, foreground ) of notification with custom style in **setBackgroundMessageHandler** but every time  I am getting both RNFN notification and custom notification. please to help me solve this issue.

**Comment by mikehardy** [comment, marked as answer]
Without posting the actual JSON you are posting to the FCM REST API no one can say anything.
If I had to guess, you are sending a notification and data chunk in your FCM JSON, so the firebase SDKs handle the notification and you are handling the data chunk by posting a notification too. You have to make a choice, one or the other really.

**Reply by LoganRaja** [reply]
index.js
`
import React from 'react';
import { AppRegistry } from 'react-native';
import App from './App';
import { name as appName } from './app.json';
import messaging from '@react-native-firebase/messaging';
import { localNotificationService } from './src/services/LocalNotificationService';
import PushNotification from 'react-native-push-notification';

messaging().setBackgroundMessageHandler(async remoteMessage => {

  console.log('Message handled in the background!', remoteMessage);

  //PushNotification.removeAllDeliveredNotifications();
   //PushNotification.clearLocalNotification({id:remoteMessage.messageId});
  let data = remoteMessage.data;
  PushNotification.localNotification({
    id: data.id,
    //autoCancel: true,
    largeIcon: 'ic_launcher',
    smallIcon: 'ic_launcher',
    bigText: data.body || '',
    subText: data.title || '',
    vibrate: true,
    vibration: 300,
    priority: 'high',
    importance: 'high', // (optional) set notification importance, default: high,
    data: data,
    title: data.title || '',
    message: data.body || '',
    playSound: true,
    soundName: 'default',
    userInteraction: false, // BOOLEAN: If the notification was opened by the user from the notification area or not
  });
  return;
});



//AppRegistry.registerHeadlessTask('RNFirebaseBackgroundMessage', () => message);
AppRegistry.registerComponent(appName, () => `App);`

package.json

"@react-native-community/push-notification-ios": "1.3.0",
    "@react-native-firebase/app": "7.3.1",
    "@react-native-firebase/messaging": "7.1.8",
    "react": "16.9.0",
    "react-native": "0.61.5",

Thanks for the reply @mikehardy , This is my code. I got remote notification perfectly, but I want to replace that remote notification with a new style but It shows new notification and default also. how to i remove or replace default notification.
(I checked android only now)

**Reply by mikehardy** [reply]
Old versions, please upgrade
You still haven't submitted the actual json you post to the FCM rest API, so I have nothing to add. My previous comment is complete
